sycl/include/CL/sycl/group_algorithm.hpp

```diff
@@ -791,14 +794,15 @@ joint_inclusive_scan(Group g, InPtr first, InPtr last, OutPtr result,
                      const ptrdiff_t &divisor) -> ptrdiff_t {
     return ((v + divisor - 1) / divisor) * divisor;
   };
-  typename InPtr::element_type x;
-  typename OutPtr::element_type carry = init;
+  typename std::remove_const<typename detail::remove_pointer<InPtr>::type>::type
```

Comment:
Why do we remove constantness here? I can be illegal.

Reply:
I did that to allow using a `const` input pointer. These two lines are used to get the underlying type, but the const qualifier is never casted away

Reply:
Without it I get `/home/michel/sycl_workspace/deploy/bin/../include/sycl/CL/sycl/group_algorithm.hpp:781:1: note: candidate template ignored: requirement 'detail::is_arithmetic<std::plus<void>>::value' was not satisfied [with Group = cl::sycl::ext::oneapi::sub_group, InPtr = const float *, OutPtr = float *, BinaryOperation = float, T = std::plus<>]
joint_inclusive_scan(Group g, InPtr first, InPtr last, OutPtr result,`

Reply:
@vladimirlaz if you want I can remove that, the point was indeed to keep const correctness

Reply:
Or I can use the type `typename detail::remove_pointer<OutPtr>::type` for `x` which shouldn't be const and avoid the issue

Reply:
I got it. You can keep it as is.

Reply:
> I got it. You can keep it as is.

Nice, you comment made me realise that I forgot some of these `remove_const`

Reply:
> I got it. You can keep it as is.

Nice, thanks!